Anton Zhukov - Solution

windings-lab opened the pull request.

windings-lab committed:
Anton Zhukov - Solution

mateacademy-ai-mentor commented on INSTRUCTION.md:
````diff
@@ -0,0 +1,46 @@
+# Test `todoapp` using port forward
+
+```
+kubectl port-forward service/todoapp-service 8080:80
````
The command `kubectl port-forward service/todoapp-service 8080:80` is incorrect because `todoapp-service` does not exist. It should target either `todoapp-nodeport` or `todoapp-clusterip` service to forward the ports correctly. Please correct it.

mateacademy-ai-mentor reviewed (changes requested):
Hey there! 👋 The configurations for your pods and services look great! However, there's a small hiccup in the `INSTRUCTION.md` file. The `kubectl port-forward` command is referencing a service named `todoapp-service` which doesn't exist. Could you please update it to use either `todoapp-nodeport` or `todoapp-clusterip` to ensure the port forwarding works correctly? Thanks! 😊

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

🤖 If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) field below and **mention me with `@mateacademy-ai-mentor` tag**. I will read your message when I perform the next review of your pull request.

windings-lab committed:
Fixed instruction.md

mateacademy-ai-mentor reviewed (approved):
Great job on setting up the todoapp with ClusterIP and NodePort services! The configuration files look good, and the instructions are clear. 🎉

Here's a summary of my review with a few suggestions:

*   **General:** Overall, the setup is well-structured and meets the requirements.

*   **INSTRUCTION.md:** The instructions for testing the todoapp are clear and cover the necessary steps, including using ClusterIP, port forwarding, and NodePort. Good job on updating the `kubectl port-forward` command with the correct service name (`todoapp-clusterip`).

*   **clusterIP.yaml:** The ClusterIP service is correctly defined, selecting pods with the `app: todolist` label and forwarding traffic to port 8080.

*   **nodeport.yaml:** The NodePort service is properly configured to expose the todoapp on port 30008, selecting pods with the correct label, and forwarding traffic to port 8080.

*   **todoapp-pod.yml:** The pod definitions for `todoapp` and `todoapp-2` look good, with correct labels, image, liveness, and readiness probes.

*   **Dockerfile:** The `Dockerfile` is not related to the task requirements, so it is not an issue.

I'm approving this PR to keep you moving forward. Keep up the great work, and don't hesitate to ask if you have any more questions!

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

windings-lab commented:
@mateacademy-ai-mentor 
Dockerfile is actually related to task:
Set all env values for the container from the pod’s manifest

I deleted ENV variable from here and added it to todoapp-pod.yml
